File: Python/other/Swap_two_nibbles.py
#Python Program to swap the two nibbles in a given byte.

#function to swap nibbles
def swapNibbles(number): 
    return ( (number & 0x0F)<<4 | (number & 0xF0)>>4 ) 

number = int(input("Enter Number: "))
print("After swapping the nibbles:")
print(swapNibbles(number)) 

'''
Sample I/O:
    Input
    Enter Number: 100

    Output:
    After swapping the nibbles:                                                                                                    
    70   

Time Complexity: O(1)
Space Complexity: O(1)

'''

File: C/other/Swap_two_nibbles.c
//algorithm for swap nibbles in O(1) time complexity and O(1) space complexity

#include <stdio.h>

int swapnibbles(int num)
{
    return (((num & 0x0F) << 4) | ((num & 0xF0) >> 4));
}

int main()
{   
    //taking input
    int num;
    printf(" Enter a number : \n");
    scanf("%d",&num);
    int result = swapnibbles(num);
    printf("output is  %d\n",result);
}

/*
Input:
Enter a number: 50
Output: 
35
*/ 
